src/diffusers/loaders/lora.py

```diff
@@ -1363,24 +1402,61 @@ def save_lora_weights(
             safe_serialization (`bool`, *optional*, defaults to `True`):
                 Whether to save the model using `safetensors` or the traditional PyTorch way with `pickle`.
         """
-        state_dict = {}
-
-        def pack_weights(layers, prefix):
-            layers_weights = layers.state_dict() if isinstance(layers, torch.nn.Module) else layers
-            layers_state_dict = {f"{prefix}.{module_name}": param for module_name, param in layers_weights.items()}
-            return layers_state_dict
+        if not USE_PEFT_BACKEND:
+            if unet_lora_config or text_encoder_lora_config or text_encoder_2_lora_config:
+                raise ValueError(
+                    "Without `peft`, passing `unet_lora_config` or `text_encoder_lora_config` or `text_encoder_2_lora_config` is not possible. Please install `peft`."
+                )
+        else:
+            from peft import LoraConfig
 
         if not (unet_lora_layers or text_encoder_lora_layers or text_encoder_2_lora_layers):
             raise ValueError(
                 "You must pass at least one of `unet_lora_layers`, `text_encoder_lora_layers` or `text_encoder_2_lora_layers`."
             )
 
+        state_dict = {}
+        metadata = {}
+
+        def pack_weights(layers, prefix, config=None):
```

Comment:
Could we keep the old `pack_weights` function and write a separate function to retrieve the local metadata?

Reply:
Done in 49a0f3ab.